packages/docusaurus-theme-classic/src/theme/DocItem/index.tsx

```diff
@@ -67,38 +67,41 @@ function DocItem(props: Props): JSX.Element {
           })}>
           <DocVersionSuggestions />
           <div className={styles.docItemContainer}>
-            <article>
+            <article className="markdown">
               {showVersionBadge && (
-                <div>
-                  <span className="badge badge--secondary">
-                    Version: {version.label}
-                  </span>
-                </div>
+                <span className="badge badge--secondary">
+                  Version: {version.label}
+                </span>
+              )}
+
+              {/*
+              Title can be declared inside md content or declared through frontmatter and added manually
+              To make both cases consistent, the added title is added under the same div.markdown block
+              See https://github.com/facebook/docusaurus/pull/4882#issuecomment-853021120
+              */}
+              {shouldAddTitle && <MainHeading>{title}</MainHeading>}
+
+              <DocContent />
+
+              {(editUrl || lastUpdatedAt || lastUpdatedBy) && (
+                <footer className={clsx('row', styles.docUpdateDetails)}>
+                  <div className="col">
+                    {editUrl && <EditThisPage editUrl={editUrl} />}
+                  </div>
+
+                  <div className={clsx('col', styles.lastUpdated)}>
+                    {(lastUpdatedAt || lastUpdatedBy) && (
+                      <LastUpdated
+                        lastUpdatedAt={lastUpdatedAt}
+                        formattedLastUpdatedAt={formattedLastUpdatedAt}
+                        lastUpdatedBy={lastUpdatedBy}
+                      />
+                    )}
+                  </div>
+                </footer>
               )}
-              <div className="markdown">
```

Comment:
was wondering if we shouldn't keep this div.markdown around the main content?

Why do we need to have article.markdown and put badge / footer under the .markdown class?

Even if we already do an exception for the title, I think this should be the only one.

Also some existing users may target .markdown to style the main content, and this may be annoying for them as main content becomes harder to target. I'd rather keep that div.markdown

Reply:
As I understand it, after we put the heading inside the div.markdown, there is less sense in this separate element.
Moreover, the targeting is usually on semantic tags (not div class), so nothing should change for search crawlers, right? https://github.com/algolia/docsearch-configs/blob/master/configs/docusaurus-2.json

But yes, given that users could apply custom styles to div.markdown element, there are possible unwanted consequences, so I bring back this tag.

Reply:
Thanks

I mean that some users might want to add extra top/bottom margins around the main div.markdown element for example. It's less convenient to do without that wrapper